cw-orch-daemon/src/builder.rs

```diff
@@ -48,6 +53,8 @@ impl DaemonAsyncBuilder {
 
     /// Set the deployment id to use for the daemon interactions
     /// Defaults to `default`
+    ///
+    /// This field is ignored for rebuilt daemon and deployment id of the original daemon used instead
     pub fn deployment_id(&mut self, deployment_id: impl Into<String>) -> &mut Self {
         self.deployment_id = Some(deployment_id.into());
         self
```

Comment:
I really want to make this editable between "rebuilt" daemons, but can't wrap my head around this

Reply:
We shouldn't put the lock in the daemon state. That way we can clone/mutate the key data and not share it between daemon instances.

Reply:
Kept lock inside DaemonState, but decreased scope to only FullAcess Daemon

Reply:
> Kept lock inside DaemonState, but decreased scope to only FullAcess DaemonState, so it looks like this now:
> 
> ```rust
> enum DaemonStateFile {
>     ReadOnly {
>         path: String,
>     },
>     FullAccess {
>         json_file_state: Arc<Mutex<JsonLockedState>>,
>     },
> }
> ```

Doesn't this allow to read from the file while the state is being written to? I.e. when creating a ReadOnly variant, is there a check that there's no lock on the file? And even if we add that check, we can still create a read-only file and then create a write one.

Reply:
> > Kept lock inside DaemonState, but decreased scope to only FullAcess DaemonState, so it looks like this now:
> > ```rust
> > enum DaemonStateFile {
> >     ReadOnly {
> >         path: String,
> >     },
> >     FullAccess {
> >         json_file_state: Arc<Mutex<JsonLockedState>>,
> >     },
> > }
> > ```
> 
> Doesn't this allow to read from the file while the state is being written to? I.e. when creating a ReadOnly variant, is there a check that there's no lock on the file? And even if we add that check, we can still create a read-only file and then create a write one.

Lock happens only for writes, we can't do anything if daemon(or something else) mutates file from another program. And dev can manage his read only daemons inside this program (i.e. read before mutations, after mutations or force write to the file to read in the process)